### ISSUE TITLE
Ordering of legend in plot_spectra

### ISSUE THREAD

**Issue by jat255** [hyperspy/hyperspy#1694]
When using the `plot_spectra` function with a padding > 0, the plots are plotted from the bottom up. When creating the legend from a list of labels, the items are added to the matplotlib legend from top down, leading to a non-intuitive labeling (especially if colors in the legend are repeated).

```python
matplotlib.style.use('seaborn-deep')
sigs = [hs.signals.Signal1D(np.arange(20)*(i/10)) for i in range(1,6)]
for s in sigs:
    s.axes_manager[0].name = 'X'
    s.axes_manager[0].units = 'arb.'
hs.plot.plot_spectra(sigs, style='cascade',
                     legend=['{}'.format(i) for i in range(1,6)])
plt.tight_layout()
```
![image](https://user-images.githubusercontent.com/1278301/28329338-7f3d66bc-6bb7-11e7-9260-c8d6f13a866b.png)

I would propose that we either plot "top-down" in `plot_spectra`, or switch the ordering of the labels in the legend (likely easier).

**Comment by jat255** [comment]
Thinking about this more, the same could be achieved by setting `padding` to `-1.0`, but I think the current "bottom-up" ordering would make most sense to users by default.

**Comment by ericpre** [comment]
Fixed in #1695.
